### PR TITLE
[node] Concurrency tests

### PR DESCRIPTION
**Pull request by rhlsthrm**
Add tests for concurrency in various situations. `packages/node/test/integration/uninstall-concurrent-mixed.spec.ts` is still not working, and is skipped for now.

### CODE REVIEW

File: packages/node/test/integration/uninstall-concurrent-mixed.spec.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        case nodeB.publicIdentifier:
+          return "Node B";
+        case nodeC.publicIdentifier:
+          return "Node C";
+        default:
+          return "unknown";
+      }
+    };
+    nodeA.on(NODE_EVENTS.PROTOCOL_MESSAGE_EVENT, (msg: ProposeMessage) => {
+      console.log(
```
Comment (LayneHaber) [inline packages/node/test/integration/uninstall-concurrent-mixed.spec.ts]: we should probably remove all this logging before putting these tests into master

File: packages/node/test/integration/uninstall-concurrent-mixed.spec.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  makeProposeCall
+} from "./utils";
+
+expect.extend({ toBeLt });
+
+jest.setTimeout(15000);
+
+// TODO: this test isnt working bc of the concurrency issues that are being investigated now.
+// reenable this test when it works
+describe.skip("Concurrently uninstalling virtual and regular applications without issue", () => {
```
Comment (LayneHaber) [inline packages/node/test/integration/uninstall-concurrent-mixed.spec.ts]: do we want to merge skipped tests into master?

Comment (snario) [reply packages/node/test/integration/uninstall-concurrent-mixed.spec.ts]: I think it's okay as we intend for this to pass

**Comment by rhlsthrm** [comment]
> Majority of these tests also have weak assertions (assertions are primarily that they will finish without throwing an explicit error. Do we want to have stronger assertions in any of these tests besides "code completes", or can we leave that to other integration/unit tests that aren't explicitly concurrency tests?

Since the other unit tests exist, I think these are fine as "weak" assertions. @LayneHaber what are the changes requested? Hesitant to remove the console logging now on the skipped test since it's being skipped anyways.

**Comment by LayneHaber** [comment]
> 
> Hesitant to remove the console logging now on the skipped test since it's being skipped anyways.

sure we can leave the `console.log`s until the test isnt skipped and were done debugging

**Pull request merged**